sense-android-library/src/nl/sense_os/datastorageengine/RealmSensor.java

```diff
@@ -0,0 +1,234 @@
+package nl.sense_os.datastorageengine;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+
+import io.realm.Realm;
+import io.realm.RealmResults;
+import nl.sense_os.service.shared.SensorDataPoint;
+
+public class RealmSensor implements Sensor {
+
+    private Realm realm = null;
+
+    private long id = -1;
+    private String name = null;
+    private String userId = null;
+    private String source = null;
+    private SensorDataPoint.DataType dataType = null;
+    private SensorOptions options = new SensorOptions();
+    private boolean synced = false;
+
+    protected RealmSensor(Realm realm, long id, String name, String userId, String source, SensorDataPoint.DataType dataType, SensorOptions options, boolean synced) {
+        this.realm = realm;
+
+        this.id = id;
+        this.name = name;
+        this.userId = userId;
+        this.source = source;
+        this.dataType = dataType;
+        this.options = options;
+        this.synced = synced;
+    }
+
+    public long getId() {
+        return id;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public String getUserId() {
+        return userId;
+    }
+
+    public String getSource() {
+        return source;
+    }
+
+    public SensorDataPoint.DataType getDataType() {
+        return dataType;
+    }
+
+    public boolean isSynced() {
+        return synced;
+    }
+
+    /**
+     * Apply options for the sensor.
+     * The fields in `options` which are `null` will be ignored.
+     * @param options
+     * @return Returns the applied options.
+     */
+    public SensorOptions setOptions (SensorOptions options) throws JSONException, DatabaseHandlerException {
+        this.options = SensorOptions.merge(this.options, options);
+        this.synced = false; // mark as dirty    TODO: is it desired behavior to set synced to false?
+
+        // store changes in the local database
+        saveChanges();
+
+        return getOptions();
+    }
+
+    /**
+     * Retrieve a clone of the current options of the sensor.
+     * @return options
+     */
+    public SensorOptions getOptions () {
+        return options.clone();
+    }
+
+    public void setSynced(boolean synced) throws DatabaseHandlerException {
+        this.synced = synced;
+
+        // store changes in the local database
+        saveChanges();
+    }
+
+    /**
+     * Insert a new DataPoint to this sensor
+     * @param value
+     * @param date
+     */
+    public void insertDataPoint(boolean value, long date) {
```

Comment:
@josdejong A mismatch of method name here. In iOS, it is insertOrUpdateDataPoint.